frontend/lib/create_listing/view_models/create_listing.viewmodel.dart

```diff
@@ -25,6 +28,85 @@ class CreateListingViewModel extends LoadingViewModel {
     _condition = condition;
     notifyListeners();
   }
+  
+  void setShowValidationErrors(bool value) {
+    _showValidationErrors = value;
+    notifyListeners();
+  }
+
+  // Validation methods
+  String? validateTitle(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Title is required';
+    }
+    if (value.length > CreateListingModel.maxTitleLength) {
+      return 'Title must be less than ${CreateListingModel.maxTitleLength} characters';
+    }
+    return null;
+  }
+
+  String? validatePrice(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Price is required';
+    }
+    
+    try {
+      final price = double.parse(value);
+      if (price <= 0) {
+        return 'Price must be greater than zero';
+      }
+      if (price > CreateListingModel.maxPrice) {
+        return 'Price cannot exceed ${CreateListingModel.maxPrice}';
+      }
+    } catch (e) {
+      return 'Please enter a valid price';
+    }
+    
+    return null;
+  }
+
+  String? validateDescription(String? value) {
+    if (value != null) {
+      if (value.length > CreateListingModel.maxDescriptionLength) {
+        return 'Description must be less than ${CreateListingModel.maxDescriptionLength} characters';
+      }
+      if (value.isNotEmpty && value.trim().isEmpty) {
+        return 'Description cannot be only whitespace';
+      }
+    }
+    return null;
+  }
+  
+  bool validateForm(GlobalKey<FormState> formKey) {
+    setShowValidationErrors(true);
+    
+    final isFormValid = formKey.currentState?.validate() ?? false;
+    final hasConditionSelected = condition.isNotEmpty;
+    
+    return isFormValid && hasImage && hasConditionSelected;
+  }
+
+  Future<bool> submitForm({
+    required GlobalKey<FormState> formKey,
+    required String title,
+    required String price,
+    required String description,
+  }) async {
+    if (!validateForm(formKey)) {
+      return false;
+    }
+    
+    try {
+      return await createListing(
+        title: title.trim(),
```

Comment:
Side comment (no action needed): we'll figure out how images are uploaded later, this is something the backend and I are still working on